contracts/nexus-gateway/src/state.rs

```diff
@@ -23,14 +31,29 @@ impl<'a> GatewayStore<'a> {
 }
 
 impl Store for GatewayStore<'_> {
-    fn store_config(&mut self, config: Config) {
-        CONFIG.save(self.storage, &config).unwrap();
+    fn save_config(&mut self, config: Config) -> Result<()> {
+        CONFIG
+            .save(self.storage, &config)
+            .change_context(ContractError::StoreFailure)
     }
 
-    fn load_config(&self) -> Config {
+    fn load_config(&self) -> Result<Config> {
         CONFIG
             .load(self.storage)
-            .expect("config must be set during contract instantiation")
```

Comment:
why this change? Feel like this should never happen.

Reply:
This could happen if deserialization fails. For example, we have some custom deserializer implemented for the Config where we do some custom validation.

Reply:
yeah but shouldn't we panic in that situation?

Reply:
panic if the error is due to a programming error, Err() if it is due to input data (or something circumstantial like unavailable services).
So if the some validation of the config fails that should be an Err

Reply:
But shouldn't we not save a config that can't be deserialized? Which would mean if we get here, then we saved a config that we shouldn't have saved, hence it would be a programming error.

My point is, I don't see how we can get to this state, where the config can't be loaded. It seems like a totally invalid contract state to me. I also don't know what can even be done on the contract if the config can't be loaded, the whole contract is basically dead at this point.

Reply:
at that point you are baking knowledge about which data came from the outside and which was created internally into the store. What if the config was never (de)serialized before? It's still very easy to convert this Err into a panic at a higher level were we have explicit knowledge about where the struct came from

Reply:
in conclusion: 
- should the contract panic if the config can't be loaded? Most likely, yes
- should the store know in which cases it's safe to panic vs returning an Err? I don't think so

Reply:
Same to me. It should be the contract instead of the store to determine when to panic.